mempool/reactor.go

```diff
@@ -129,6 +142,21 @@ func (memR *Reactor) Receive(e p2p.Envelope) {
 	// broadcasting happens from go routines per peer
 }
 
+func (memR *Reactor) EnableInOutTxs() {
```

Comment:
Easier still, why not just:

```suggestion
func (memR *Reactor) EnableInOutTxs() {
	if !memR.WaitSync() {
		return
	}
```

Reply:
This works if we assume that we never transition back, right? I mean, wait sync is true to then become false forever.

Even in this case, what exactly happens if `AddPeer` and this method are called at the same time?

Reply:
> This works if we assume that we never transition back, right?

That's the current assumption, isn't it? `SwitchToConsensus` in the consensus reactor works like this.

> what exactly happens if AddPeer and this method are called at the same time?

What about the following code?
```suggestion
func (memR *Reactor) EnableInOutTxs() {
	// If waitSync is not true, it means that txs are already enabled, then
	// return immediately; otherwise, set it to false and enable txs.
	if !memR.waitSync.CompareAndSwap(true, false) {
		return
	}
```

Reply:
All these comments were addressed with the introduction of `waitSyncCh`.